script/c511027014.lua

```diff
@@ -0,0 +1,71 @@
+--ハイドライブ・リ・アドベント
+--Hydradrive Readvent
+--Scripted by Playmaker 772211
+local s,id=GetID()
+function s.initial_effect(c)
+	--Activate
+	local e1=Effect.CreateEffect(c)
+	e1:SetCategory(CATEGORY_SPECIAL_SUMMON)
+	e1:SetType(EFFECT_TYPE_ACTIVATE)
+	e1:SetProperty(EFFECT_FLAG_CARD_TARGET)
+	e1:SetCode(EVENT_FREE_CHAIN)
+	e1:SetTarget(s.target)
+	e1:SetOperation(s.activate)
+	c:RegisterEffect(e1)
+end
+function s.cfilter(c)
+	return c:IsFaceup()
+end
+function s.filter(c,e,tp,atk)
+	return c:IsType(TYPE_LINK) and c:IsAttackBelow(atk) and c:IsCanBeSpecialSummoned(e,0,tp,false,false)
+end
+function s.target(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
+	local cg=Duel.GetMatchingGroup(s.cfilter,tp,0,LOCATION_MZONE,nil)

```

Comment:
Since the filter only contain one function, you can include it directly in the `GetMatchingGroup` function, that should look something like that
```suggestion
	local cg=Duel.GetMatchingGroup(Card.IsFaceup,tp,0,LOCATION_MZONE,nil)
```